src/hooks/useColorGenerator/useColorGenerator.ts

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable valid-jsdoc */
+import {useThemeType} from '../../components/theme/useThemeType';
+
+import {getPersistentColor, getTextColor} from './color';
+import type {UseColorGeneratorProps} from './types';
+
+/**
+ * It is used to create a unique color from a token (string) and to obtain an inverted color (black or white), 
+ * ensuring higher text contrast compared to the current color, which is usually better for human perception.
+ *
+ * Usage example:
+ ```tsx
+    import React from 'react';
+    import {Avatar} from '@gravity-ui/uikit';
+
+    const Component = ({ token, text, ...avatarProps }) => {
+        const {color, textColor} = useColorGenerator({
+            token,
+        });
+
+        return (
+            <Avatar
+                {...avatarProps}
+                text={text}
+                color={text ? textColor : undefined}
+                backgroundColor={color}
+            />
+        );
+    };
+```
+*/
+export function useColorGenerator(props: UseColorGeneratorProps) {
+    const theme = useThemeType();
+
+    const {color, hue, saturation, lightness} = getPersistentColor({
+        ...props,
+        theme,
+    });
+
+    const textColor = getTextColor(hue, saturation, lightness);
```

Comment:
Let's wrap them both in `React.useMemo`, to save computation time during rerenders

Reply:
i think useMemo will actually consume more memory and computations here. 
all props (intensity and seed) will be listed in the useMemo dependencies, and the functions will always have to be recalculated when any prop changes. So useMemo will be redundant here.

there is also no point in wrapping `getTextColor` in useMemo, because if the hue, saturation or lightness color parameters changed, it means the props have changed.

To avoid unnecessary re-renders, the user should wrap their component in memo.

Reply:
If props of the hook not changing why we should rerender? It's a rule of thumb, to return memoized vaue from the hooks

Reply:
why will the values be recalculated if the props haven't changed?
in my project, i wrapped the component in `memo`, which uses similar hook to prevent unnecessary re-renders.
in this case, excessive memoization in the hook will only make things worse.
but if that's the standart practice in gravity, then ok

Reply:
Every component's render will recalculate hook's value. Suggesting users to memoize their components instead of memoizing inside the hook is better? I don't think so